engine/runtime-compiler/src/main/java/org/enso/compiler/pass/MiniPassChainer.java

```diff
@@ -0,0 +1,60 @@
+package org.enso.compiler.pass;
+
+import java.util.List;
+import org.enso.compiler.core.IR;
+import org.enso.compiler.pass.IRPass.IRMetadata;
+
+/** Utility class for chaining mini passes together. */
+public final class MiniPassChainer {
+  private MiniPassChainer() {}
+
+  /**
+   * Chains a list of mini passes into a single pass that applies them in order.
+   *
+   * @param passes Ordered list of mini passes.
+   * @return Chained mini pass that applies all the provided mini passes in the given order.
+   */
+  public static MiniIRPass chain(List<MiniIRPass> passes) {
+    return new MiniIRPass() {
+      @Override
+      public MiniIRPass prepare(IR ir) {
+        throw new UnsupportedOperationException("unimplemented");
```

Comment:
This will have to be implemented one day:
```java
        List<MiniIRPass> newMiniPasses = null;
        for (var miniPass : passes) {
          var nextPass = miniPass.prepare(ir);
          if (nextPass != miniPass) {
            newMiniPasses = new ArrayList<>();
            for (var copy : passes) {
              if (copy == miniPass) {
                 break;
              }
              newMiniPasses.add(copy);
            }
            if (newMiniPasses != null) {
               newMiniPasses.add(nextPass);
            }
          }
        }
        return newMiniPasses == null ? this : chain(newMiniPasses);
```

Reply:
Fixed in 8be8505fd4